core/services/pipeline/orm.go

```diff
@@ -383,19 +381,20 @@ VALUES
 	(:pipeline_spec_id, :pruning_key, :meta, :all_errors, :fatal_errors, :inputs, :outputs, :created_at, :finished_at, :state) 
 RETURNING id
 	`
-		rows, errQ := sqlx.NamedQueryContext(ctx, tx.ds, pipelineRunsQuery, runs)
-		if errQ != nil {
-			return errors.Wrap(errQ, "inserting finished pipeline runs")
-		}
-		defer rows.Close()
 
 		var runIDs []int64
-		for rows.Next() {
-			var runID int64
-			if errS := rows.Scan(&runID); errS != nil {
-				return errors.Wrap(errS, "scanning pipeline runs id row")
+		err := sqlutil.NamedQueryContext(ctx, tx.ds, pipelineRunsQuery, runs, func(rows *sqlx.Rows) error {
+			for rows.Next() {
+				var runID int64
+				if errS := rows.Scan(&runID); errS != nil {
+					return errors.Wrap(errS, "scanning pipeline runs id row")
+				}
+				runIDs = append(runIDs, runID)
```

Comment:
This API could be tightened up:
```suggestion
		err := sqlutil.NamedQueryContext(ctx, tx.ds, pipelineRunsQuery, runs, func(scanRow func(any) error}) error {
			var runID int64
			if errS := rows.Scan(&runID); errS != nil {
				return errors.Wrap(errS, "scanning pipeline runs id row")
			}
			runIDs = append(runIDs, runID)
```
As long as this is not too restrictive :thinking:

Reply:
Done